### PR TITLE
[Wave] Add GQA/MQA Vanilla template

### PR DESCRIPTION
**Pull request by harsh-nod**
This PR creates a new template for GQA/MQA vanilla attention that supports both causal masks and sliding windows for both f16 and f8 datatypes.

### CODE REVIEW

File: tests/kernel/wave/attention/vanilla_attention_test.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def testAttentionCausal(
 
         def sliding_window_mask(q_seq_length, kv_seq_length, window_size):
             mask = device_ones((q_seq_length, kv_seq_length), dtype=torch.bool)
-            mask = mask.tril().triu(-sliding_window)
+            mask = mask.tril().triu(-window_size)
```
Comment (stanwinata) [inline tests/kernel/wave/attention/vanilla_attention_test.py:301]: nice! this is a minor bug fix I reckon?

File: tests/kernel/wave/common/utils.py [hunk 2]
```diff
@@ -75,14 +76,33 @@ def scaled_dot_product_attention_bhsd(
...
 
+    if sliding_window >= 0:
+        assert is_causal, f"Sliding window only supported with causal"
+
     if is_causal:
         seq_len_q, seq_len_k = attn_logits.shape[-2], attn_logits.shape[-1]
-        causal_mask: Tensor = torch.tril(
-            torch.ones(
-                (seq_len_q, seq_len_k), device=attn_logits.device, dtype=torch.bool
+        if sliding_window < 0:
```
Comment (raikonenfnu) [inline tests/kernel/wave/common/utils.py:84]: any thoughts on dropping the if-else condition in favor of `if (sliding_window >= 0) existing_mask.triu(-window_size)`?

Comment (harsh-nod) [reply tests/kernel/wave/common/utils.py:84]: yeah good point, that will simplify the code.

**Review by stanwinata — APPROVED** [review]
LGTM!

**Review by raikonenfnu — APPROVED** [review]
LGTM! (sorry for double ping, reviewing from my mac which has my old GH)

**Comment by harsh-nod** [comment]
Thanks, I just added the FP8 variant but it doesnt fully work yet. I will ping again when it is ready. Thanks for taking a look!

**Comment by harsh-nod** [comment]
@raikonenfnu - can you take another look? F8 is now passing.

File: iree/turbine/kernel/wave/templates/gqa_vanilla_attention.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    # FP8 offset
+    # If we need to truncate to fp8 post softmax we apply a scaling to use the
+    # full fp8 range. We can do this with a offset as post `exp2` this equates
+    # to multiplying by a static value. We are able to do this as `max` and
+    # `sum` are scaled by the same value so the end result is the same.
+    FP8_OFFSET_VAL = math.log2(F8_MAX / ATTENTION_SOFTMAX_MAX)
+
+    # Dequant Tensor Scaling
+    DEQUANT_QK = q_scale * k_scale
+    if use_fp8:
```
Comment (raikonenfnu) [inline iree/turbine/kernel/wave/templates/gqa_vanilla_attention.py:63]: A TODO for coming PRs is to guard more stuff with the `use_fp8` flag, I reckon we don't to apply FP8_OFFSET_VAL when it's no fp8, as well as the v_scale(i.e someone can theoretically turn on v_scale but not fp8).

Comment (harsh-nod) [reply iree/turbine/kernel/wave/templates/gqa_vanilla_attention.py:63]: good point, will fix.

Comment (harsh-nod) [reply iree/turbine/kernel/wave/templates/gqa_vanilla_attention.py:63]: I am surprised that this passed the fp16 tests.
